app/Http/Controllers/AdminClinic/AppointmentController.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Http\Controllers\AdminClinic;
+
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Appointment;
+use App\Clinic;
+
+class AppointmentController extends Controller
+{
+
+    /**
+     * Display a listing of appointment of clinic.
+     *
+     * @param String $slug slug
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index($slug)
```

Comment:
không cần phải đưa slug vào mỗi function trong controller như thế này. 
Tạo 1 cái `BaseController` cho `AdminClinic`, trong `__constructor` của class đó, lấy `slug` từ request ra, query để lấy được clinic từ slug, rồi gắn vào property của class luôn. 
ở mỗi class con chỉ cần dùng `$this->clinic` là được.